src/cdk-experimental/column-resize/resize-strategy.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Inject, Injectable, OnDestroy, Provider} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {coerceCssPixelValue} from '@angular/cdk/coercion';
+
+import {ColumnResize} from './column-resize';
+
+/**
+ * Provides an implementation for resizing a column.
+ * The details of how resizing works for tables for flex mat-tables are quite different.
+ */
+@Injectable()
+export abstract class ResizeStrategy {
+  abstract applyColumnSize(
+      cssFriendlyColumnName: string,
+      columnHeader: HTMLElement,
+      sizeInPx: number): void;
+
+  abstract applyMinColumnSize(
+      cssFriendlyColumnName: string,
+      columnHeader: HTMLElement,
+      minSizeInPx: number): void;
+
+  abstract applyMaxColumnSize(
+      cssFriendlyColumnName: string,
+      columnHeader: HTMLElement,
+      minSizeInPx: number): void;
+}
+
+/**
+ * The optimially performing resize strategy for &lt;table&gt; elements with table-layout: fixed.
+ * Tested against and outperformed:
+ *   CSS selector
+ *   CSS selector w/ CSS variable
+ *   Updating all cell nodes
+ */
+@Injectable()
+export class TableLayoutFixedResizeStrategy extends ResizeStrategy {
+  applyColumnSize(_: string, columnHeader: HTMLElement, sizeInPx: number): void {
+    columnHeader.style.width = coerceCssPixelValue(sizeInPx);
+  }
+
+  applyMinColumnSize(_: string, columnHeader: HTMLElement, sizeInPx: number): void {
+    columnHeader.style.minWidth = coerceCssPixelValue(sizeInPx);
+  }
+
+  applyMaxColumnSize(): void {
+    // Intentionally omitted as max-width causes strange rendering issues in Chrome.
+    // Max size will still apply when the user is resizing this column.
+  }
+}
+
+/**
+ * The optimally performing resize strategy for flex mat-tables.
+ * Tested against and outperformed:
+ *   CSS selector w/ CSS variable
+ *   Updating all mat-cell nodes
```

Comment:
Were these tests something persistent that we could reference in the future, or just something local/transient?

Reply:
I just ran them locally in the development app with 100 and 500 rows.

I could imagine a unit test that recreates that scenario, but I'd rather not be the one to write it :P